x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/expression/function/scalar/between/BetweenBaseTestCase.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.eql.expression.function.scalar.between;
+
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.concurrent.Callable;
+
+public abstract class BetweenBaseTestCase extends ESTestCase {
```

Comment:
I'll let other reviewers weigh in on the necessity of this approach here.

That isn't an usual approach with randomized testing. The Elasticsearch infra in CI does run the tests continuously and having a randomized value for parameters is usually enough.
A more common approach is to introduce randomness in tests with various `random*` methods and leave CI do its thing. A personal approach (that others I believe are following) is to stress test such a random test class with `-Dtests.iters=1000` or such a large value before creating the PR. I'm not saying this is enough, but it gives one a greater degree of confidence over the committed code.

Reply:
This seems to be the approach utilized by serializations tests, it's just extracted into the base class there, which we could consolidate across functions tests. Easy to remove. Let me know.